worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/regen/PaperweightRegen.java

```diff
@@ -683,4 +700,22 @@ public CompletableFuture<?> processChunk(Long xz, List<ChunkAccess> accessibleCh
 
     }
 
+    private class SingleBiomeProvider extends BiomeProvider {
```

Comment:
Would it make sense to move this class to Regenerator and reuse it? Also the provider selection code (the if else on the option) could be extracted then.

Reply:
It would